girder-dandi-archive/tests/test_rest.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+
+from girder_dandi_archive.util import create_drafts_collection
+
+from girder.models.collection import Collection
+from girder.models.folder import Folder
+
+from pytest_girder.assertions import assertStatus, assertStatusOk
+
+
+@pytest.fixture
+def draftsFolders(db):
+    red_herring_collection = Collection().createCollection(
+        "red_herring_collection", reuseExisting=True
+    )
+    # A folder that matches dandiset metadata, but not in drafts collection.
+    red_herring_dandiset_000000_folder = Folder().createFolder(
+        parent=red_herring_collection,
+        parentType="collection",
+        name="000000",
+        public=True,
+    )
+    meta = {
+        "dandiset": {
+            "name": "red",
+            "description": "herring",
+            "identifier": "000000",
+            "version": "draft",
+        }
+    }
+    Folder().setMetadata(red_herring_dandiset_000000_folder, meta)
+    return_dict = {
+        "red_herring_collection": red_herring_collection,
+        "red_herring_folder": red_herring_dandiset_000000_folder,
+    }
+    return_dict["drafts_collection"] = create_drafts_collection()
+    yield return_dict
+
+
+@pytest.mark.plugin("dandi_archive")
+def testCreateDandiset(server, draftsFolders, user):
+    drafts_collection_id = str(draftsFolders["drafts_collection"]["_id"])
+
+    resp = server.request(
+        path="/dandi",
+        method="POST",
+        user=user,
+        params={
+            "name": "test dandiset 0 name",
+            "description": "test dandiset 0 description",
+        },
+    )
+    assertStatusOk(resp)
+    test_dandiset_0_folder = resp.json
+    test_dandiset_0_folder_meta = test_dandiset_0_folder["meta"]["dandiset"]
+
+    assert drafts_collection_id == test_dandiset_0_folder["parentId"]
+    assert "000000" == test_dandiset_0_folder["name"]
+    assert "000000" == test_dandiset_0_folder_meta["identifier"]
+    assert "test dandiset 0 name" == test_dandiset_0_folder_meta["name"]
+    assert "test dandiset 0 description" == test_dandiset_0_folder_meta["description"]
+    assert "draft" == test_dandiset_0_folder_meta["version"]
+
+    resp = server.request(
```

Comment:
Technically, this is a separate test case, and it should live in a separate function. Given that it's trying to test stateful behavior (creation when at least one other Dandiset exists), you could initialize it by either artificially incrementing `DANDISET_IDENTIFIER_COUNTER` at the start of the case, or alternatively (and less artificial) you could make a `dandiset` fixture to ensure you start the case with another Dandiset already in the database.

Practically, I don't think you *need* to change this, but it's a good habit to make much smaller, more granular test cases that follow the pattern:
```
# set up
# execute
# assert
```
rather than large test cases which attempt to pack multiple executions into one.